### PR TITLE
age_population_year #9

### PR DESCRIPTION
**Pull request by aarathybabu97**

### CODE REVIEW

**Comment by aarathybabu97** [comment]
```r
library(conmat)
 world_data <- socialmixr::wpp_age()
 age_population(
  data = world_data,
 country_col = country,
 country = "Asia",
  age_col = lower.age.limit,
  year_col = year,
  year = 2010
)%>%
   dplyr::arrange(lower.age.limit)%>%
   head(5)
#>   country year population lower.age.limit upper.age.limit
#> 1    Asia 2010  360336208               0               4
#> 2    Asia 2010  350780197               5               9
#> 3    Asia 2010  351260914              10              14
#> 4    Asia 2010  372841496              15              19
#> 5    Asia 2010  386404948              20              24

age_population(
  data = world_data,
  age_col = lower.age.limit,
  year_col = year,
  year = 2010
)%>%
  dplyr::arrange(lower.age.limit)%>%
  head(5)
#>       country year population lower.age.limit upper.age.limit
#> 1 Afghanistan 2010    5011447               0               4
#> 2      Africa 2010  165738403               0               4
#> 3     Albania 2010     158681               0               4
#> 4     Algeria 2010    3844982               0               4
#> 5      Angola 2010    4162211               0               4

age_population(
  data = world_data,
  country_col = country,
  country = "Australia",
  age_col = lower.age.limit
)%>%
  dplyr::arrange(lower.age.limit)%>%
  head(5)
#>     country year population lower.age.limit upper.age.limit
#> 1 Australia 1950     888741               0               4
#> 2 Australia 1955     993468               0               4
#> 3 Australia 1960    1094108               0               4
#> 4 Australia 1965    1163156               0               4
#> 5 Australia 1970    1247078               0               4

Created on 2022-02-18 by the [reprex package](https://reprex.tidyverse.org/) (v2.0.1)
```

Note: 85+ ages are kept under 85-120 age group

File: R/check_args_age_population_year.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  
+  if (!is.null(country)) {
+    age_population_country_df <- age_population_year_df %>%
+      dplyr::filter({{ country_col }} == {{ country }})
+    
+    return(age_population_country_df)
+    
+  }
+  else
+    return(age_population_year_df)
```
Comment (njtierney) [inline R/check_args_age_population_year.R:23]: Add braces around `else`:

```
else {
return(...)
}
```

File: R/age-population-year.R [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     age_var <- dplyr::pull(data, {{ age_col }})
+    
+    
+    if (is.numeric(age_var)) {
+      label <- c(paste(seq(0, 84, by = 5),
+                       seq(0 + 5 - 1, 85 - 1, by = 5),
+                       sep = "-"), paste(85, 120, sep = "-"))
+      
+      data <- data %>%
+        dplyr::mutate(age_group = cut({{ age_col }},
```
Comment (njtierney) [inline R/age-population-year.R:54]: Add a linebreak here:

```
dplyr::mutate(
  age_group = ..,
  breaks = ...
  ...
  )
```

File: R/get_age_population_function.R [hunk 3]
```diff
@@ -1,42 +1,43 @@
...
+      bin_width = bin_widths(  {{ age_col }} ),
+      midpoint =  {{ age_col }} + bin_width / 2,
       # scaling down the population appropriately
       log_pop = log(population / bin_width)
     )
   
   # find the maximum of the bounded age groups, and the populations above and
   # below
-  max_bound <- max(pop_model$lower.age.limit)
+  max_bound <- max(pop_model%>%
```
Comment (njtierney) [inline R/get_age_population_function.R:33]: move `pop_model` inside of `dplyr::pull`:

```
dplyr::pull(
  pop_model,
  {{ age_col }}
  )

```

File: R/separate_age_group.R [hunk 4]
```diff
@@ -0,0 +1,12 @@
+separate_age_group <- function(data, age_col) {
+  return(
+    data %>%
+      tidyr::separate({{age_col}}, c("lower.age.limit", "upper.age.limit"), "[[:punct:]]", extra = "merge") %>%
```
Comment (njtierney) [inline R/separate_age_group.R:4]: Add line breaks and. indentation here:

```
tidyr::seprate(
  {{ age_col }},
 ...,
)
```

File: conmat.Rproj [hunk 5]
```diff
@@ -1,8 +1,8 @@
 Version: 1.0
 
 RestoreWorkspace: Default
-SaveWorkspace: Default
-AlwaysSaveHistory: Default
+SaveWorkspace: Ask
+AlwaysSaveHistory: No
```
Comment (njtierney) [inline conmat.Rproj:5]: Not sure how this changed - let's discuss

File: inst/old_age_population_year.R [hunk 6]
```diff
@@ -0,0 +1,24 @@
+
+age_population_year <- function(data, age_col, year_col, year) {
+  age_var <- dplyr::pull(data, {{ age_col }})
+  .args <- as.list(match.call()[-1])
```
Comment (njtierney) [inline inst/old_age_population_year.R:4]: I don't think `.args` is needed anymore?

**Review by njtierney — CHANGES_REQUESTED** [review]

**Comment by njtierney** [comment]
Can you also ensure that these functions return a `tibble` instead of a `data.frame`

File: R/get_age_population_function.R [hunk 7]
```diff
@@ -1,42 +1,45 @@
...
 #' \dontrun{
-#' if (interactive()) {
-#'   # EXAMPLE1
-#' }
+#' get_age_population_function(population, age_col = lower.age.limit)
 #' }
 #' @export
-get_age_population_function <- function(population) {
+get_age_population_function <- function(population, 
+                                        age_col=lower.age.limit) {
```
Comment (njtierney) [inline R/get_age_population_function.R:17]: I remember you had made some changes to this, using `.data`, but I can't seem to find where those changes are in the other PRs

**Review by njtierney — CHANGES_REQUESTED** [review]

**Review by njtierney — APPROVED** [review]
Looks. good to me!

**Pull request merged**